learning.py

```diff
@@ -0,0 +1,81 @@
+import nn_architecture
+import data_loader as dl
+
+import logging
+import numpy as np
+import copy
+
+import torch
+from torch import nn
+import torch.optim as optim
+from torch.optim import lr_scheduler
+from torch.utils.data import Dataset, DataLoader
+
+
+def train(model, x, y, matrix):
+    best_loss = np.inf
+    loss_history = []
+
+    # Iterate throughout the epochs
+    for epoch in range(num_epochs):
```

Comment:
The dataset is virtually unlimited In our case.
So, I don't think its necessary to limit ourselves to the concept of epoch.
That is, to show the network same pair of scans 100 times.
Maybe, we could make little fewer repetitions on each pair, but show more diverse set of pairs, using the same amount of compute?

Reply:
need to discuss this and add to the next update